src/NUnitFramework/framework/Internal/Execution/EventQueue.cs

```diff
@@ -1,11 +1,25 @@
 // Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
 
+using System;
 using System.Collections.Concurrent;
 using System.Threading;
 using NUnit.Framework.Interfaces;
 
 namespace NUnit.Framework.Internal.Execution
 {
+    /// <summary>
+    /// Interface for ALL event types that can be queued for processing.
+    /// </summary>
+    /// <typeparam name="TListener"></typeparam>
+    public interface IEvent<in TListener>
+    {
+        /// <summary>
+        /// The Send method is implemented by derived classes to send the event to the specified listener.
+        /// </summary>
+        /// <param name="listener">The listener.</param>
+        void Send(TListener listener);
```

Comment:
Can you explain this? Events sending themselves sounds not logical. What about multiple listeners?

Reply:
This is the way it has been for years. No change here.  I agree it doesn't look right, but I am not changing the logic here.
The code the Send method in a concrete event like TestStartedEvent looks like
```csharp
public override void Send(ITestListener listener)
{
    listener.TestStarted(_test);
}
```
where _test is an ITest.  So it is actually not sending itself, but the content it holds. An Event is just an envelope.

Reply:
And yes, multiple listeners are not implemented.

Reply:
Thanks for the explanation